### PR TITLE
perf: Avoid Vector2 creation in `Sprite.render`

### PR DESCRIPTION
**Pull request by spydon**
# Description

This creates two new private vector2 objects that can be re-used inside of `Sprite.render` so that we don't have to create new objects in there.

## Checklist

- [x] I have followed the [Contributor Guide] when preparing my PR.
- [x] I have updated/added tests for ALL new/updated/fixed functionality.
- [x] I have updated/added relevant documentation in `docs` and added dartdoc comments with `///`.
- [x] I have updated/added relevant examples in `examples` or `docs`.

## Breaking Change?

- [x] No, this PR is not a breaking change.

## Related Issues

[Contributor Guide]: https://github.com/flame-engine/flame/blob/main/CONTRIBUTING.md
[Conventional Commit]: https://conventionalcommits.org
[CHANGELOG]: https://github.com/flame-engine/flame/blob/main/CHANGELOG.md

### CODE REVIEW

File: packages/flame/lib/src/sprite.dart [hunk 1]
```diff
@@ -97,18 +101,30 @@ class Sprite {
...
+      _tmpRenderPosition.setFrom(position);
+    } else {
+      _tmpRenderPosition.setZero();
+    }
+
+    if (size != null) {
+      _tmpRenderSize.setFrom(size);
+    } else {
+      _tmpRenderSize.setFrom(srcSize);
+    }
```
Comment (st-pasha) [inline packages/flame/lib/src/sprite.dart:114]:
```suggestion
    _tmpRenderSize.setFrom(size ?? srcSize);
```

File: packages/flame/lib/src/sprite.dart [hunk 2]
```diff
@@ -81,6 +81,10 @@ class Sprite {
     );
   }
 
+  // Used to avoid the creation of new Vector2 objects in render.
+  late final _tmpRenderPosition = Vector2.zero();
+  late final _tmpRenderSize = Vector2.zero();
```
Comment (st-pasha) [inline packages/flame/lib/src/sprite.dart:86]: Probably no need to make them `late`.

**Review by wolfenrain — APPROVED** [review]

File: packages/flame/lib/src/sprite.dart (hunk 1 above)
Comment (spydon) [reply packages/flame/lib/src/sprite.dart:114]: Lol, really is time for me to go to bed :joy: 
Thanks!

File: packages/flame/lib/src/sprite.dart (hunk 2 above)
Comment (spydon) [reply packages/flame/lib/src/sprite.dart:86]: No harm in doing so right? If some user has some big collection of sprites that aren't being rendered yet they have a little bit less overhead.

Comment (st-pasha) [reply packages/flame/lib/src/sprite.dart:86]: Well, the overhead is that there needs to be a runtime check whether the `late` variable has actually been initialized or not. And I'm not sure whether such a check is debug-mode only, or if it is also done in production...

Comment (spydon) [reply packages/flame/lib/src/sprite.dart:86]: I don't think such a check is even done in debug-mode? Then we'd have better stack traces for `LateInitializationError` right?

Comment (spydon) [reply packages/flame/lib/src/sprite.dart:86]: That must be very minimal? But I don't know how it is done.
I'd say it is worth the trade-off of not carrying around two `Vector2` before they are used.
I can do some benchmarking tomorrow.

Comment (st-pasha) [reply packages/flame/lib/src/sprite.dart:86]: Another possibility is to declare them `static`. Then you have only one object instance no matter how many sprites

Comment (spydon) [reply packages/flame/lib/src/sprite.dart:86]: True, single thread ftw 😄

Comment (spydon) [inline packages/flame/lib/src/sprite.dart:86]:
```suggestion
  static final _tmpRenderPosition = Vector2.zero();
  static final _tmpRenderSize = Vector2.zero();
```

**Pull request merged**